Fix Y-sorted root item having modulation applied twice

kleonc opened the pull request:
Fixes #95667.

For the y-sorted subtree's root modulation is obtained once again after gathering y-sorted items, just like e.g. transform. In the similar manner it needs to be cancelled out.
 (again fixing my own mistake, from #78134 :upside_down_face:)

clayjohn reviewed (approved):
Should be fine to merge after a rebase

akien-mga commented:
@kleonc Would you be able to rebase this soon?

kleonc committed:
Fix Y-sorted root item having modulation applied twice

kleonc commented:
Rebased, and fixed the change as the previous `/ modulate` would cancel out the whole modulation (including the modulation part propagated from the parent non-y-sorted items). Instead, only the single item's (y-sorted subtree root's) modulation is cancelled out now with `/ ci->modulate`.

But before merging: @clayjohn is it fine as is or should we handle potential division by zero? :thinking:

Repiteo commented:
Thanks!
